File: go-sdk/pkg/models/shared/timelineskillcard.go
package shared

import (
	"time"
)

// TimelineSkillCardEntriesAppears
// The timestamp for an entry.
type TimelineSkillCardEntriesAppears struct {
	End   *int64 `json:"end,omitempty"`
	Start *int64 `json:"start,omitempty"`
}

// TimelineSkillCardEntries
// An single item that's placed on multiple items on the timeline.
type TimelineSkillCardEntries struct {
	Appears  []TimelineSkillCardEntriesAppears `json:"appears,omitempty"`
	ImageURL *string                           `json:"image_url,omitempty"`
	Text     *string                           `json:"text,omitempty"`
}

type TimelineSkillCardInvocationTypeEnum string

const (
	TimelineSkillCardInvocationTypeEnumSkillInvocation TimelineSkillCardInvocationTypeEnum = "skill_invocation"
)

// TimelineSkillCardInvocation
// The invocation of this service, used to track
// which instance of a service applied the metadata.
type TimelineSkillCardInvocation struct {
	ID   string                              `json:"id"`
	Type TimelineSkillCardInvocationTypeEnum `json:"type"`
}

type TimelineSkillCardSkillTypeEnum string

const (
	TimelineSkillCardSkillTypeEnumService TimelineSkillCardSkillTypeEnum = "service"
)

// TimelineSkillCardSkill
// The service that applied this metadata.
type TimelineSkillCardSkill struct {
	ID   string                         `json:"id"`
	Type TimelineSkillCardSkillTypeEnum `json:"type"`
}

// TimelineSkillCardSkillCardTitle
// The title of the card.
type TimelineSkillCardSkillCardTitle struct {
	Code    *string `json:"code,omitempty"`
	Message string  `json:"message"`
}

type TimelineSkillCardSkillCardTypeEnum string

const (
	TimelineSkillCardSkillCardTypeEnumTimeline TimelineSkillCardSkillCardTypeEnum = "timeline"
)

type TimelineSkillCardTypeEnum string

const (
	TimelineSkillCardTypeEnumSkillCard TimelineSkillCardTypeEnum = "skill_card"
)

// TimelineSkillCard
// A Box Skill metadata card that places a list of images on a
// timeline.
type TimelineSkillCard struct {
	CreatedAt      *time.Time                         `json:"created_at,omitempty"`
	Duration       *int64                             `json:"duration,omitempty"`
	Entries        []TimelineSkillCardEntries         `json:"entries"`
	Invocation     TimelineSkillCardInvocation        `json:"invocation"`
	Skill          TimelineSkillCardSkill             `json:"skill"`
	SkillCardTitle *TimelineSkillCardSkillCardTitle   `json:"skill_card_title,omitempty"`
	SkillCardType  TimelineSkillCardSkillCardTypeEnum `json:"skill_card_type"`
	Type           TimelineSkillCardTypeEnum          `json:"type"`
}

File: java-sdk/lib/src/main/java/models/shared/TimelineSkillCard.java
package .models.shared;

import java.time.OffsetDateTime;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import .utils.DateTimeSerializer;
import .utils.DateTimeDeserializer;

/**
 * TimelineSkillCard
 * A Box Skill metadata card that places a list of images on a
 * timeline.
**/
public class TimelineSkillCard {
    @JsonInclude(Include.NON_ABSENT)
    @JsonSerialize(contentUsing = DateTimeSerializer.class)
    @JsonDeserialize(contentUsing = DateTimeDeserializer.class)
    @JsonProperty("created_at")
    public OffsetDateTime createdAt;
    public TimelineSkillCard withCreatedAt(OffsetDateTime createdAt) {
        this.createdAt = createdAt;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("duration")
    public Long duration;
    public TimelineSkillCard withDuration(Long duration) {
        this.duration = duration;
        return this;
    }
    @JsonProperty("entries")
    public TimelineSkillCardEntries[] entries;
    public TimelineSkillCard withEntries(TimelineSkillCardEntries[] entries) {
        this.entries = entries;
        return this;
    }
    @JsonProperty("invocation")
    public TimelineSkillCardInvocation invocation;
    public TimelineSkillCard withInvocation(TimelineSkillCardInvocation invocation) {
        this.invocation = invocation;
        return this;
    }
    @JsonProperty("skill")
    public TimelineSkillCardSkill skill;
    public TimelineSkillCard withSkill(TimelineSkillCardSkill skill) {
        this.skill = skill;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("skill_card_title")
    public TimelineSkillCardSkillCardTitle skillCardTitle;
    public TimelineSkillCard withSkillCardTitle(TimelineSkillCardSkillCardTitle skillCardTitle) {
        this.skillCardTitle = skillCardTitle;
        return this;
    }
    @JsonProperty("skill_card_type")
    public TimelineSkillCardSkillCardTypeEnum skillCardType;
    public TimelineSkillCard withSkillCardType(TimelineSkillCardSkillCardTypeEnum skillCardType) {
        this.skillCardType = skillCardType;
        return this;
    }
    @JsonProperty("type")
    public TimelineSkillCardTypeEnum type;
    public TimelineSkillCard withType(TimelineSkillCardTypeEnum type) {
        this.type = type;
        return this;
    }
}